src/graph/validator/MatchValidator.cpp

```diff
@@ -550,13 +560,15 @@ Status MatchValidator::validateWith(const WithClause *with,
       auto label = static_cast<const LabelExpression *>(col->expr())->name();
       auto found = withClauseCtx.yield->aliasesAvailable.find(label);
       DCHECK(found != withClauseCtx.yield->aliasesAvailable.end());
-      if (!withClauseCtx.aliasesGenerated.emplace(col->alias(), found->second).second) {
+      /*if (!withClauseCtx.aliasesGenerated.emplace(col->alias(), found->second).second) {
         return Status::SemanticError("`%s': Redefined alias", col->alias().c_str());
-      }
+      }*/
+      withClauseCtx.aliasesGenerated.emplace(col->alias(), found->second);
```

Comment:
We should differentiate the variable aliases defined in with clause and inherited from the former clause, maybe MATCH, YIELD whatever. We only need to check whether the alias defined in with clause has been defined in current clause or former clause.

Reply:
Understood. My changes here are wrong. Reverted.